src/tools/ScenarioMeasurement/Startup/Perfcollect.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Reflection;
+
+namespace ScenarioMeasurement
+{
+    public class PerfCollect : IDisposable
+    {
+        private readonly string startupDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        private ProcessHelper perfCollectProcess;
+        public string TraceName { get; private set; }
+        public string TraceDirectory { get; private set; }
+        public string TraceFilePath { get; private set; }
+        private List<KernelKeyword> KernelEvents = new List<KernelKeyword>();
+        private List<ClrKeyword> ClrEvents = new List<ClrKeyword>();
+        public PerfCollect(string traceName, Logger logger) : this(traceName, Environment.CurrentDirectory, logger)
+        {
+        }
+
+        public PerfCollect(string traceName, string traceDirectory, Logger logger)
+        {
+            TraceName = traceName;
+            string perfCollectScript = Path.Combine(startupDirectory, "perfcollect");
+            if (!File.Exists(perfCollectScript))
+            {
+                throw new FileNotFoundException($"Pefcollect not found at {perfCollectScript}. Please rebuild the project to download it.");
+            }
+
+            if (String.IsNullOrEmpty(traceName))
+            {
+                throw new ArgumentException("Trace file name cannot be empty.");
+            }
+
+
+            if (!Directory.Exists(traceDirectory))
+            {
+                Directory.CreateDirectory(traceDirectory);
+            }
+            TraceDirectory = traceDirectory;
+
+
+            perfCollectProcess = new ProcessHelper(logger)
+            {
+                ProcessWillExit = true,
+                Executable = perfCollectScript,
+                Timeout = 300
+            };
+        }
+
+        public ProcessHelper.Result Start()
+        {
+            string arguments = $"start {TraceName} -events ";
```

Comment:
It doesn't meaningfully matter here, but this is a good pattern to think about using `StringBuilder` for. Every time you append the string you end up allocating a new string. In high performance code that's definitely a pattern to avoid. Just FYI.